### PR TITLE
Workflowtemp1 (#396)

### PR DESCRIPTION
**Pull request by matthewchungcodefresh**
* Workflow-template-jirascan

* Update README.md

* Update workflowTemplate.yaml

### CODE REVIEW

File: examples/jira-scan-workflow/versions/0.0.1/docs/jira-sonar-template.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+* COMMENT_BODY (optional) - Text to add to the comment
+* VERBOSE (optional) - Enable verbose logging by setting to true
+* REVISION (optional) - the revision to checkout. defaults to main
+* IMAGE_NAME (required) - The image name to give to the built image
+* DOCKERFILE (optional) - The path to your dockerfile. defaults to Dockerfile
+* CONTEXT (optional) - the context of file system that will be passed. defaults to .
+* TAG (optional) - The tag that will be given to the image. defaults to latest
+* CF_API_KEY (required) - The Kubernetes secret containing the Codefresh API key created by runtime
+* CF_API_KEY_SECRET_KEY (optional) - The key in the Kubernetes secret that has the Codefresh API key created by runtime. Default is 'token'
+* IMAGE_SHA (required) - The image sha that was imported into Codefresh to enrich
```
Comment (lrochette) [inline examples/jira-scan-workflow/versions/0.0.1/docs/jira-sonar-template.md:35]: this should not be a parameter, it is returned by image-report-info
one does not know the SHA image at the start of a build

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+          #   kubectl create secret generic sonar-secret \
+          #   --from-literal=token=$SONAR_TOKEN --dry-run=client \
+          #   --save-config -o yaml | kubectl apply -f - -n $NAMESPACE
+          - name: SONAR_LOGIN
+            default: sonar-secret
+          - name: SONAR_LOGIN_SECRET_KEY
+            default: token
+          - name: COMMENT_BODY
+            default: ''
+          - name: value-to-promote
```
Comment (lrochette) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:105]: it's strange to mix upper and lower case for parameters, stick to 1 format

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+  entrypoint: main
+  templates:
+    - name: main
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Builds a docker image using Kaniko, creates and updates a jira through each step, scans and tests the image repo, and deploys using a canary rollout'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/workflows/jira-scan-workflow/assets/icon.svg"
+          argo-hub-template/icon_background: "#f4f4f4"
+      inputs:
+        parameters:
```
Comment (lrochette) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:33]: it's a bunch load of parameters

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+          # https://codefresh.io/argohub/workflow-template/sonar
+          - name: scan-repo
+            dependencies: [test-image]
+            templateRef:
+              name: argo-hub.sonar.0.0.1
+              template: sonar-scanner
+            arguments:
+              artifacts:
+                - name: REPO
+                  from: "{{tasks.clone-step-repo.outputs.artifacts.REPO}}"
```
Comment (lrochette) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:300]: this should be "tasks.clone-step-repo.outputs.artifacts.repo" in lower case as defined in the block

`      outputs:
        artifacts:
          - name: repo
            path: /tmp/repo
`
in the clone template
How were you able to make it run?

**Review by lrochette — CHANGES_REQUESTED** [review]

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+  annotations:
+    argo-hub/version: '0.0.1'
+    argo-hub/description: 'Workflow template for a full deployment using jira, sonar and a canary rollout'
+    argo-hub/categories: 'argo'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Matthew Chung'
+    argo-hub/owner_email: 'matthew.chung@codefresh.io'
+    argo-hub/owner_avatar: 'https://github.com/matthewchungcodefresh.png'
+    argo-hub/owner_url: 'https://github.com/matthewchungcodefresh'
+    argo-hub/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:14]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 6]
```diff
@@ -0,0 +1,562 @@
...
+          - key: .dockerconfigjson
+            path: config.json
+        secretName: '{{ inputs.parameters.DOCKER_CONFIG_SECRET }}'
+  entrypoint: main
+  templates:
+    - name: main
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Builds a docker image using Kaniko, creates and updates a jira through each step, scans and tests the image repo, and deploys using a canary rollout'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:30]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 7]
```diff
@@ -0,0 +1,562 @@
...
+                  value: 'Completed'
+                - name: VERBOSE
+                  value: '{{ inputs.parameters.VERBOSE }}'
+
+    # test template
+    - name: test
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Test image pulling correctly and printing it content files'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:494]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 8]
```diff
@@ -0,0 +1,562 @@
...
+        image: '{{ inputs.parameters.IMAGE }}'
+        command: [sh, -c]
+        args: ["ls"]
+
+    # canary rollout template 10%
+    - name: rollout-10
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Applies a 10% canary rollout'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:509]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 9]
```diff
@@ -0,0 +1,562 @@
...
+          - /cf-deploy-kubernetes ./canary-manual-approval/service-preview.yaml
+          - /cf-deploy-kubernetes ./canary-manual-approval/service-all.yaml  
+          - /cf-deploy-kubernetes ./canary-manual-approval/rollout.yaml   
+
+    # canary rollout template 33%
+    - name: rollout-33
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Applies a 33% canary rollout'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:531]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

File: examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml [hunk 10]
```diff
@@ -0,0 +1,562 @@
...
+        workingDir: /tmp/repo
+        commands:
+          - /app/kubectl-argo-rollouts-linux-amd64 promote golang-sample-app-deployment -n canary --context "mydemoAkscluster@BizSpark Plus"
+    
+    # canary rollout template 100%
+    - name: rollout-100
+      metadata:
+        annotations:
+          argo-hub-template/description: 'Applies a 100% canary rollout'
+          argo-hub-template/icon_url: "https://github.com/codefresh-io/argo-hub/blob/main/examples/jira-scan-workflow/assets/icon.svg"
```
Comment (eti-codefresh) [inline examples/jira-scan-workflow/versions/0.0.1/workflowTemplate.yaml:550]: need to be change to this "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/examples/jira-scan-workflow/assets/icon.svg"

**Review by eti-codefresh — CHANGES_REQUESTED** [review]

**Review by lrochette — APPROVED** [review]

**Review by eti-codefresh — APPROVED** [review]

**Pull request merged**